kube-runtime/src/utils/predicate.rs

```diff
@@ -114,6 +115,33 @@ where
     }
 }
 
+/// Configuration for predicate filtering with cache TTL
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// Time-to-live for cache entries
+    ///
+    /// Entries older than this duration will be evicted from the cache.
+    /// Default is 1 hour.
+    pub ttl: Duration,
```

Comment:
Let's make this private and have a `ttl` setter function instead (as is common elsewhere, like the controller::Config). Then you can use `Config::default().ttl(duration)` and we are free to add fields without this being a breaking change.